docs/book/src/topics/failure-domains/worker-nodes.md

```diff
@@ -0,0 +1,132 @@
+# Failure domains in worker nodes
+
+To ensure that the worker machines are spread across failure domains, we need to create N `MachineDeployment` for your N failure domains, scaling them independently. Resiliency to failures comes from having multiple `MachineDeployment`.
```

Comment:
Do we want to elaborate here how .failureDomain and .Subnet relate?
e.g 
We have 2 sources for subnets:
1. If subnet.id or subnet.filters are specified, we directly query AWS
2. All other cases use the subnets provided in the cluster network spec without ever calling AWS

Relates to https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2864/
cc @codablock

Reply:
I have added that in [control-planes](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6200cb4f95d4a746ddad637565b1e18bea7cd2d1/docs/book/src/topics/failure-domains/control-planes.md). Do we want to add similar explanation for worker nodes as well?

Reply:
ah thanks @Ankitasw, I had missed that. My +1 to present it in a way that's clear that apply to both cp and worker nodes as well.

Reply:
Modified a note [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/61fdf9fa1303f9d471da3e321926d65cd8e542b5/docs/book/src/topics/failure-domains/control-planes.md?plain=1#L75), so that we know it applied for worker nodes as well